resources/src/effect/mod.rs

```diff
@@ -0,0 +1,564 @@
+use amount::Amount;
+use asset::{AssetIdentifier, Flag};
+use serde::{de, Deserialize, Deserializer};
+
+pub mod account;
+pub mod signer;
+pub mod trustline;
+mod trade;
+
+pub use self::trade::Trade;
+
+#[cfg(test)]
+mod test;
+
+/// A successful operation will yield zero or more effects. These effects represent specific
+/// changes that occur in the ledger, but are not necessarily directly reflected in the ledger or
+/// history, as transactions and operations are.
+#[derive(Debug)]
+pub struct Effect {
+    id: String,
+    paging_token: String,
+    kind: Kind,
+}
+
+/// Each effect type is representing by a kind and captures data specific to that
+/// type within it's newtype.
+#[derive(Debug, Deserialize)]
+pub enum EffectKind {
+    /// A collection of effects that represent updates to an account
+    Account(account::Kind),
+    /// A collection of effects that represent updates to an account signer
+    Signer(signer::Kind),
+    /// A collection of effects that represent updates to a trustline
+    Trustline(trustline::Kind),
+    /// An effect representing a trade being executed
+    Trade(Trade),
```

Comment:
Perhaps we should use a `Kind` here as well so that if an effect was added for trades that it would be extensible without breaking the public interface?